### PR TITLE
Add Comparator for Strings

### PR DESCRIPTION
**Pull request by pwsdotru**
Add one separate class for string compare

### CODE REVIEW

**Comment by pwsdotru** [comment]
@sebastianbergmann can you please see this PR

I think it will good base for add canonicalize for string. 

But need solve how calculate canonicalize.

File: src/StringComparator.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        $expectedToCompare = $expected;
+        $actualToCompare   = $actual;
+
+        if ($canonicalize) {
+            $expectedToCompare = $this->canonicalizeString($expectedToCompare);
+            $actualToCompare   = $this->canonicalizeString($actualToCompare);
+        }
+
+        if ($ignoreCase) {
+            $expectedToCompare = \strtolower($expectedToCompare);
```
Comment (u-mulder) [inline src/StringComparator.php:54]: `strtolower` has no idea how to lower case anything accept 0 t0 127 ascii codes. Run your tests with `CVÓamión` for example.

**Comment by sebastianbergmann** [comment]
This branch has conflicts that must be resolved. Also, please explain what you want to achieve with this. So far, I do not see any benefit and am inclined to reject this pull request.

**Comment by pwsdotru** [comment]
@sebastianbergmann main idea was separtate comporator for scalar and string types. Because for string need make canonicalize comporators. 
Else need add condition like is_string and it code will not very clean

**Pull request closed without merging**